conda/base/context.py

```diff
@@ -522,6 +523,13 @@ def __init__(self, search_path=None, argparse_args=None, **kwargs):
         self._set_env_vars(APP_NAME)
         self._set_argparse_args(argparse_args)
 
+    def add_environment_file_config_source(self, file, data):
```

Comment:
In addition to configuration coming from argparse args, environment vars and `.condarc` config files, `environment.yaml` files provide some amount of configuration. For now, `environment.yaml` is only able to provide channel configuration. But in the future, I expect that users might be able to specify more config for their environment. 
This aims to provide a way to inject configuration from the environment file into the running context.

Reply:
Good point, can do!